### PR TITLE
fix(api): updates the api to standard naming and adds docs for selection

### PR DESCRIPTION
**Pull request by amcdnl**
This PR updates the change event in `SelectionModel` from `onChange` to `changed` to match the new api standard. It also adds basic docs for the selection model utility.

### CODE REVIEW

File: src/cdk/collections/collections.md [hunk 1]
```diff
@@ -1,3 +1,34 @@
-### Collections
+The `collections` package provides a set of utilities for managing collections.
 
-A set of utilities for managing collections.
\ No newline at end of file
+### `SelectionModel`
+`SelectionModel` is a utility for powering selection of one or more options from a list.
+This utility is used in components such as the selection list, table selections and chip lists.
```
Comment (jelbourn) [inline src/cdk/collections/collections.md]: Something like
```markdown
`SelectionModel` stores the selection state for components that support multiple selection.
```

The word "utility" isn't a good fit here

File: src/cdk/collections/selection.ts [hunk 2]
```diff
@@ -34,7 +34,13 @@ export class SelectionModel<T> {
   }
 
   /** Event emitted when the value has changed. */
-  onChange: Subject<SelectionChange<T>> | null = this._emitChanges ? new Subject() : null;
+  changed: Subject<SelectionChange<T>> | null = this._emitChanges ? new Subject() : null;
```
Comment (jelbourn) [inline src/cdk/collections/selection.ts]: Didn't we land on `change` present-tense?

File: src/cdk/collections/collections.md (hunk 1 above)
Comment (amcdnl) [reply src/cdk/collections/collections.md]: How about `model`?

File: src/cdk/collections/selection.ts (hunk 2 above)
Comment (amcdnl) [reply src/cdk/collections/selection.ts]: We landed on `myEventThingChange` but the rest of them ended up past tense.

File: src/cdk/collections/collections.md (hunk 1 above)
Comment (jelbourn) [reply src/cdk/collections/collections.md]: That works

File: src/cdk/collections/selection.ts (hunk 2 above)
Comment (jelbourn) [reply src/cdk/collections/selection.ts]: Which things are `changed`?

Comment (amcdnl) [reply src/cdk/collections/selection.ts]: A few I know of `selectedChanged`, `_scrollDistanceChanged` and `_selectedIndexChanged`. Let me know what you want to do.

**Comment by amcdnl** [comment]
bump @jelbourn

File: src/cdk/collections/selection.ts [hunk 3]
```diff
@@ -34,7 +34,13 @@ export class SelectionModel<T> {
...
+
+  /**
+   * Event emitted when the value has changed.
+   * @deprecated Use `changed` instead.
```
Comment (jelbourn) [inline src/cdk/collections/selection.ts]: Need to add a deletion-target here (8.0)

Comment (amcdnl) [reply src/cdk/collections/selection.ts]: Done :)

**Comment by jelbourn** [comment]
Commit message fix-up: should use `feat(collections)` and also mention the API deprecation

**Comment by jelbourn** [comment]
@amcdnl  github shows this PR as having no conflicts, but our presubmit script thinks it does. Could you rebase it?

**Pull request merged**